### ISSUE TITLE
feature request: Filter / own domain/reverse of Filter Proposal

### ISSUE THREAD

**Issue by PotatoCarl** [RealRaven2000/quickFilters#271]
When creating filter by domain/address with the filter assistant, I sometimes grab "my own mail" respectively want to generate a filter from the "sent" folder.
In that case, as sender domain my own domain/email address is used.

It would make sense to detect the sender/account domain and address and in such a case "inverse" the search, i.e. grab the domain/email from the recipient.

**Comment by RealRaven2000** [comment]
Yeah that one is tricky - that's why I show email details in the bottom of the assistant, it is a little harder to guess what the user wants. With the normal (filter by from) I usually completely drop emails that are sent by oneself. (say you pull in a bunch of emails from a thread and there are some of your own emails in there, obviously you do not want to filter out everything you sent yourself if you are basing a filter on a "from" address. But generally I am not convinced that making the domain filter more automated is such a safe thing - you can very quickly end up filtering all outlook.com / gmail.com if you're not checking the domains manually and put some interactive thought into it.

While it may be an advantage for super power users who build 10 or more filters every day, I am not sure that the effort will be worth it for the majority of users. If I change that behavior for everyone, you would have some obscured rule that -from the perspective of an ordinary user - arbitrarily grabs some domain and end up with filters that do what they don't want/

**Comment by PotatoCarl** [comment]
Okay, I found an email where it happens:

![grafik](https://github.com/user-attachments/assets/abc8433e-6813-4fea-bd3e-ce765dd9cafc)

![grafik](https://github.com/user-attachments/assets/6f1747cf-5b1e-468f-b1d5-baf2e868c383)


![grafik](https://github.com/user-attachments/assets/12b64b4a-3cf9-45be-a3c2-03192b4bd7b0)

This is well reproducible with this email

**Comment by RealRaven2000** [comment]
Can you export the template ("Nach Domain") - I believe that is a custom filter template? at least searching my code base doesn't turn up any matches  - and the email (as eml file), send off list if you like. Maybe add [quickFilters] issue #271 to the subject line...

**Comment by PotatoCarl** [comment]
```json
{
  "accountName": "mailbox://nobody@Local%20Folders",
  "rootFolderURL": "mailbox:///XXXXXXXXXXXXXXXXXX/Mail/Local%20Folders/",
  "date": "2024-10-09T07:30:07.441Z",
  "filters": [
    {
      "filterName": "quickFilterCustomTemplate: Nach Domain",
      "filterDesc": "",
      "filterType": 81,
      "temporary": false,
      "actionCount": 1,
      "enabled": true,
      "actionList": [
        {
          "type": 2,
          "priority": 4
        }
      ],
      "searchTerms": [
        {
          "attrib": 1,
          "value": {
            "attrib": 1,
            "str": "@%from(domain)%"
          },
          "booleanAnd": false,
          "beginsGrouping": false,
          "endsGrouping": false
        },
        {
          "attrib": 6,
          "value": {
            "attrib": 6,
            "str": "@%from(domain)%"
          },
          "booleanAnd": false,
          "beginsGrouping": false,
          "endsGrouping": false
        }
      ]
    }
  ]
}
```
